api_v3/services/EntryServerNodeService.php

```diff
@@ -54,16 +54,28 @@ private function addNewEntryServerNode(KalturaEntryServerNode $entryServerNode)
 	 * @action update
 	 * @param int $id
 	 * @param KalturaEntryServerNode $entryServerNode
+	 * @param int $duration
 	 * @return KalturaEntryServerNode|null|object
 	 * @throws KalturaAPIException
 	 */
-	public function updateAction($id, KalturaEntryServerNode $entryServerNode)
+	public function updateAction($id, KalturaEntryServerNode $entryServerNode, $duration = 0)
```

Comment:
remove duration from API.
Add KalturaEntryServerNodeRecordingInfo container on entry server node that can hold all the related recording input and outputs.